src/main/java/KeyVaultManagedStorageSamples.java

```diff
@@ -34,72 +31,102 @@ public KeyVaultManagedStorageSamples() throws InterruptedException, ExecutionExc
-            if (definition.roleName().equals("Storage Account Key Operator Service Role")) {
-                keyVaultRole = definition;
-                break;
-            }
+        Optional<RoleDefinition> operationDefinition =
+                azureResourceManager.accessManagement()
+                        .roleDefinitions().listByScope("/")
+                        .stream()
+                        .filter(roleDefinition -> "Storage Account Key Operator Service Role".equals(roleDefinition.roleName()))
+                        .findFirst();
+        if (operationDefinition.isPresent()) {
+            azureResourceManager.accessManagement().roleAssignments()
+                    .define(UUID.randomUUID().toString()) //Needs to be a UUID formatted String
+                    .forObjectId("93c27d83-f79b-4cb2-8dd4-4aa716542e74") //This is the Azure Key Vault Service Principal
+                    .withRoleDefinition(operationDefinition.get().id())
+                    .withScope(storageAccount.id())
+                    .create();
         }
 
-        azure.accessManagement().roleAssignments()
-                .define(UUID.randomUUID().toString()) //Needs to be a UUID formatted String
-                .forObjectId("93c27d83-f79b-4cb2-8dd4-4aa716542e74") //This is the Azure Key Vault Service Principal
-                .withRoleDefinition(keyVaultRole.id())
-                .withScope(storageAccount.id())
-                .create();
-
-        // Setting the storage acocunt can only be called by a user account with access to the keys of the storage account.
-        // Therefore, we grant the user that created the storage account access to the vault
-        // as well as the storage account.
-
         System.out.println("Creating new vault");
-        Vault vault = azure.vaults().define(VAULT_NAME)
+        vault = azureResourceManager
+                .vaults().define(VAULT_NAME)
                 .withRegion(VAULT_REGION)
                 .withExistingResourceGroup(RESOURCE_GROUP)
                 .defineAccessPolicy()
-                    .forObjectId(USER_OID)
-                    .allowSecretAllPermissions()
-                    .allowStorageAllPermissions()
-                    .attach()
-                .withDeploymentDisabled()
+                .forObjectId(USER_OID)
+                .allowSecretAllPermissions()
+                .allowStorageAllPermissions()
+                .allowCertificateAllPermissions()
+                .allowKeyAllPermissions()
+                .attach()
+                .defineAccessPolicy()
+                .forObjectId(storageAccount.innerModel().identity().principalId())
+                .allowKeyPermissions(KeyPermissions.UNWRAP_KEY,KeyPermissions.WRAP_KEY,KeyPermissions.GET)
+                .attach()
+                .withPurgeProtectionEnabled()
+                .withSoftDeleteEnabled()
                 .create();
         VAULT_URI = vault.vaultUri();
-
+        Key keyVaultKey = vault.keys()

```

Comment:
It looks like you are using the management plane library to create a key in place of the data plane Key Vault Keys library. Is that the intended scenario this sample wants to demonstrate?

Reply:
Yes, as you might have understood, using the management plane library to create a key in place of the data plane Key Vault Keys library.